develop/items/custom-enchantment-effects.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Custom Enchantment Effects
+description: Learn how to create your enchantment effects.
+authors:
+  - krizh-p
+---
+
+# Custom Enchantments {#custom-enchantments}
+
+Since 1.21, custom enchantments have taken a "data-driven" approach; while this makes adding simple enchantments (ie. increasing attack damage) more straightforward and easy, it also complicates creating complex enchantments. This was done by breaking down what enchantments do into effect components. An effect component contains code for what special things an enchantment should do. By default, Minecraft supports various effects such as item damage, knockback, experience, and more--however, this guide will focus on creating custom enchantment effects that are not supported by default. **[Before contuining, it is heavily suggested you first determine if the default Minecraft effects will work for your usecase.](https://minecraft.wiki/w/Enchantment_definition#Effect_components)** The rest of the guide will assume you understand how "simple" data-driven enchantments are configured.
```

Comment:
Please break this up into multiple lines. It's a bit difficult to read.
![image](https://github.com/FabricMC/fabric-docs/assets/93472213/5eb80f5e-b68d-4e56-bf56-749e4a786236)